### PR TITLE
Timer activity throws null error exception #6240

### LINKED ISSUE #6240: Timer activity throws null error exception
That happens in dev branch.

Error log:
_Orchard.Tasks.BackgroundService - (null) - Error while processing background task "Orchard.Workflows.Activities.TimerBackgroundTask" on tenant "Default". [(null)]
System.NullReferenceException: Object reference not set to an instance of an object.
   at Orchard.Workflows.Activities.TimerBackgroundTask.Sweep() in d:\sites\orchard sites\rpa-mu.ru-dev\src\Orchard.Web\Modules\Orchard.Workflows\Activities\TimerActivity.cs:line 132
   at Orchard.Tasks.BackgroundService.Sweep()_

It happens on line (where ContentItemRecord is null)
`var contentItem = _contentManager.Get(action.WorkflowRecord.ContentItemRecord.Id, VersionOptions.Latest);`

### PR DESCRIPTION
**Pull request by infofromca**
fix #6240, #6445

### CODE REVIEW

**Comment by sebastienros** [comment]
I don't think you should write the tenant name explicitly as it is already included automatically by the logging infrastructure.

**Comment by sebastienros** [comment]
Fix is good. Can you squash your commits?

**Comment by sebastienros** [comment]
And force push to the same branch/PR, don't need to create a new one. Link on its way ...

**Comment by BenedekFarkas** [comment]
How to squash (explanation from Sébastien from the last meeting): https://youtu.be/Dq6iagiOXiw?t=51m19s

**Comment by jtkech** [comment]
I was going to suggest to only do the following without stopping the workflow:

    var contentItem = action.WorkflowRecord.ContentItemRecord != null ?
        _contentManager.Get(action.WorkflowRecord.ContentItemRecord.Id, VersionOptions.Latest) : null;

Otherwise, if the previous activity doesn't provide a content item to the flow (as a form submit without the CreateContent option), the workflow will be in the running state indefinitively.

Best

**Comment by infofromca** [comment]
@sebastienros  took @jtkech 's suggestion and Squashed

**Comment by sebastienros** [comment]
Your last commit broke the formatting. And then please re-squash.

**Comment by infofromca** [comment]
@sebastienros re-squashed it

File: src/Orchard.Web/Modules/Orchard.Workflows/Activities/TimerActivity.cs [hunk 1]
```diff
@@ -122,19 +124,29 @@ public class TimerBackgroundTask : IBackgroundTask {
...
 
+        public ILogger Logger { get; set; }
         public void Sweep() {
             var awaiting = _awaitingActivityRepository.Table.Where(x => x.ActivityRecord.Name == "Timer").ToList();
             
             
             foreach (var action in awaiting) {
-                var contentItem = _contentManager.Get(action.WorkflowRecord.ContentItemRecord.Id, VersionOptions.Latest);
+                try { 
+                var contentItem = action.WorkflowRecord.ContentItemRecord != null ? _contentManager.Get(action.WorkflowRecord.ContentItemRecord.Id, VersionOptions.Latest) : null;
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Workflows/Activities/TimerActivity.cs]: Please indent the lines here

**Comment by sebastienros** [comment]
Please target 1.10.x

**Pull request closed without merging**